### PR TITLE
fix(a11y): Empty buttons

### PR DESCRIPTION
**Pull request by tille**
- Adds missing aria-label to empty button (ellipsis for comments-menu)

### CODE REVIEW

File: src/elements/content-sidebar/activity-feed/comment/Comment.js [hunk 1]
```diff
@@ -140,6 +143,7 @@ class Comment extends React.Component<Props, State> {
                                 <Media.Menu
```
Comment (jstoffan) [inline src/elements/content-sidebar/activity-feed/comment/Comment.js:143]: It looks like all `Media.Menu` components will suffer this same fate. Should we add a required `label` prop to that component and fix AppActivity and Tasks, as well?

File: src/elements/content-sidebar/activity-feed/comment/Comment.js [hunk 2]
```diff
@@ -140,6 +143,7 @@ class Comment extends React.Component<Props, State> {
...
                                     isDisabled={isConfirmingDelete}
                                     data-testid="comment-actions-menu"
+                                    aria-label={commentOptionsAriaLabel}
```
Comment (jstoffan) [inline src/elements/content-sidebar/activity-feed/comment/Comment.js:146]: This is minor, but we try to alphabetize component props, object properties, variable declarations, etc. when possible.

File: src/elements/content-sidebar/activity-feed/comment/Comment.js [hunk 3]
```diff
@@ -111,13 +112,15 @@ class Comment extends React.Component<Props, State> {
...
             onEdit,
+            intl,
         } = this.props;
         const { isConfirmingDelete, isEditing, isInputOpen } = this.state;
         const createdAtTimestamp = new Date(created_at).getTime();
         const createdByUser = created_by || PLACEHOLDER_USER;
         const canEdit = onEdit !== noop && permissions.can_edit;
         const canDelete = permissions.can_delete;
         const isMenuVisible = (canDelete || canEdit) && !isPending;
+        const commentOptionsAriaLabel = intl.formatMessage(messages.commentOptionsArialLabel);
```
Comment (jstoffan) [inline src/elements/content-sidebar/activity-feed/comment/Comment.js:123]: Can we inline this since it's only used once?

File: src/elements/content-sidebar/activity-feed/comment/__tests__/__snapshots__/Comment.test.js.snap [hunk 4]
```diff
@@ -81,135 +81,139 @@ exports[`elements/content-sidebar/ActivityFeed/comment/Comment should correctly
 `;
 
 exports[`elements/content-sidebar/ActivityFeed/comment/Comment should correctly render comment when translation is enabled 1`] = `
-<ForwardRef(ActivityCard)
```
Comment (jstoffan) [inline src/elements/content-sidebar/activity-feed/comment/__tests__/__snapshots__/Comment.test.js.snap:84]: These snapshots changes are a little concerning. Do we need to `.dive()` or mock out `react-intl` in the test to get into the component again?

File: src/elements/content-sidebar/activity-feed/comment/Comment.js [hunk 5]
```diff
@@ -225,4 +229,4 @@ class Comment extends React.Component<Props, State> {
     }
 }
 
-export default Comment;
+export default injectIntl(Comment);
```
Comment (jstoffan) [inline src/elements/content-sidebar/activity-feed/comment/Comment.js:232]: It may be safer to move the entire menu to its own component and wrap that with `injectIntl`, since this seems to have had some consequences for the parent component.

Comment (tille) [reply src/elements/content-sidebar/activity-feed/comment/Comment.js:232]: did not understand this comment
may you elaborate, please

Comment (tille) [reply src/elements/content-sidebar/activity-feed/comment/Comment.js:232]: do you mean to create something like CommentMenu component then import it and use it here?

Comment (jstoffan) [reply src/elements/content-sidebar/activity-feed/comment/Comment.js:232]: Yep, that was my initial idea for a solution if `injectIntl` is interfering with refs/callbacks as the snapshots imply.

File: src/elements/content-sidebar/activity-feed/comment/__tests__/__snapshots__/Comment.test.js.snap (hunk 4 above)
Comment (tille) [reply src/elements/content-sidebar/activity-feed/comment/__tests__/__snapshots__/Comment.test.js.snap:84]: I'll need a hand with this specs 
maybe you can point me to a file with `react-intl` mocked or an example of how to deal with `injectIntl`

File: src/elements/content-sidebar/activity-feed/comment/Comment.js (hunk 5 above)
Comment (tille) [reply src/elements/content-sidebar/activity-feed/comment/Comment.js:232]: you are completely right but we are running out of time since first deliverable to webAIM have to be send for tomorrow.
so we are going to close this issue for the time being and resume after the receive feedback from webAIM.

**Comment by fsuarez-globant** [comment]
This PR will be closed for the time being and resume after the receive feedback from webAIM.

File: src/elements/content-sidebar/activity-feed/comment/Comment.js (hunk 1 above)
Comment (alexkrolick) [reply src/elements/content-sidebar/activity-feed/comment/Comment.js:143]: We could probably default it to "Options" (i18n string), since that's almost always what the "..." menu will represent

File: src/components/media/MediaMenu.tsx [hunk 6]
```diff
@@ -33,19 +42,25 @@ const MediaMenu = ({
...
-        >
-            <IconEllipsis color={bdlGray50} height={16} width={16} />
-        </PlainButton>
-        <Menu {...menuProps}>{children}</Menu>
-    </DropdownMenu>
-);
-
-export default MediaMenu;
+}: MediaMenuProps) => {
+    const commentOptionsAriaLabel = intl.formatMessage(messages.commentOptionsArialLabel);
```
Comment (jstoffan) [inline src/components/media/MediaMenu.tsx:48]: Can we inline this call with `aria-label` to allow this component to remain an implicit return function?

File: src/components/media/MediaMenu.tsx [hunk 7]
```diff
@@ -10,7 +11,15 @@ import { Menu } from '../menu';
 import { bdlGray50 } from '../../styles/variables';
 import './Media.scss';
 
-export interface MediaMenuProps extends PlainButtonProps {
+const messages = defineMessages({
+    commentOptionsArialLabel: {
+        id: 'be.contentSidebar.activityFeed.comment.commentOptionsArialLabel',
```
Comment (jstoffan) [inline src/components/media/MediaMenu.tsx:16]: Can we update the `id` property of this message should reflect its new host component, since it's not directly related to the activity feed now?

File: src/components/media/MediaMenu.tsx [hunk 8]
```diff
@@ -10,7 +11,15 @@ import { Menu } from '../menu';
 import { bdlGray50 } from '../../styles/variables';
 import './Media.scss';
 
-export interface MediaMenuProps extends PlainButtonProps {
+const messages = defineMessages({
```
Comment (jstoffan) [inline src/components/media/MediaMenu.tsx:14]: Can we move this to a `messages.js` file in this directory?

File: src/elements/content-sidebar/activity-feed/comment/messages.js [hunk 9]
```diff
@@ -27,6 +27,11 @@ const messages = defineMessages({
         defaultMessage: '{time, date, full} at {time, time, short}',
         description: 'Comment posted full date time for title',
     },
+    commentOptionsArialLabel: {
```
Comment (jstoffan) [inline src/elements/content-sidebar/activity-feed/comment/messages.js:30]: Is this message needed?

File: src/elements/content-sidebar/activity-feed/comment/__tests__/__snapshots__/Comment.test.js.snap [hunk 10]
```diff
@@ -212,4 +212,4 @@ exports[`elements/content-sidebar/ActivityFeed/comment/Comment should render com
     </MediaBody>
   </Media>
 </ForwardRef(ActivityCard)>
-`;
+`;
```
Comment (jstoffan) [inline src/elements/content-sidebar/activity-feed/comment/__tests__/__snapshots__/Comment.test.js.snap:215]: Can we revert this change?

File: src/components/media/MediaMenu.tsx (hunk 8 above)
Comment (tille) [reply src/components/media/MediaMenu.tsx:14]: do you really want a whole new file for a single variable?
I take this file as reference https://github.com/box/box-ui-elements/blob/master/src/components/time-input/TimeInput.tsx#L15

File: src/components/media/MediaMenu.tsx (hunk 7 above)
Comment (alexkrolick) [reply src/components/media/MediaMenu.tsx:16]:
```suggestion
        id: 'boxui.media.menuButtonArialLabel',
        ```

File: src/components/media/MediaMenu.tsx (hunk 8 above)
Comment (jstoffan) [reply src/components/media/MediaMenu.tsx:14]: It's less about my personal preference than it is the convention for most (though not all) components in the repository.

Comment (tille) [reply src/components/media/MediaMenu.tsx:14]: okas, no problem
just asking because that other file 

going to move it into `messages.js`

File: src/components/media/MediaMenu.tsx (hunk 7 above)
Comment (tille) [reply src/components/media/MediaMenu.tsx:16]: thank you @alexkrolick

File: src/components/media/messages.tsx [hunk 11]
```diff
@@ -0,0 +1,16 @@
+/**
+ * @file i18n messages
+ * @author Box
+ */
+
+import { defineMessages } from 'react-intl';
+
+const messages = defineMessages({
+    commentOptionsArialLabel: {
```
Comment (jstoffan) [inline src/components/media/messages.tsx:9]: Should this be `menuButtonArialLabel`, as well?

File: src/components/media/MediaMenu.tsx [hunk 12]
```diff
@@ -33,13 +35,15 @@ const MediaMenu = ({
...
     menuProps = {},
+    intl,
     ...rest
 }: MediaMenuProps) => (
     <DropdownMenu constrainToScrollParent isRightAligned {...dropdownProps}>
         <PlainButton
             isDisabled={isDisabled}
             className={classnames('bdl-Media-menu', className)}
             type={ButtonType.BUTTON}
+            aria-label={intl.formatMessage(messages.commentOptionsArialLabel)}
```
Comment (jstoffan) [inline src/components/media/MediaMenu.tsx:46]: Very minor, but we usually try to alphabetize props, object properties, etc. when possible.

**Review by jstoffan — APPROVED** [review]
Thanks for your patience while we iterated on this! LGTM.

File: src/components/media/MediaMenu.tsx (hunk 12 above)
Comment (tille) [reply src/components/media/MediaMenu.tsx:46]: but don't know how it will look after alphabetize this property

Comment (tille) [reply src/components/media/MediaMenu.tsx:46]: you mean to move aria-label to be the first prop?

**Review by jstoffan — APPROVED** [review]
